python-package/lightgbm/dask.py

```diff
@@ -102,20 +109,30 @@ def _train(client, data, label, params, model_factory, weight=None, **kwargs):
     y : dask array of shape = [n_samples]
         The target values (class labels in classification, real numbers in regression).
     params : dict
-    model_factory : lightgbm.LGBMClassifier or lightgbm.LGBMRegressor class
+    model_factory : lightgbm.LGBMClassifier, lightgbm.LGBMRegressor, or lightgbm.LGBMRanker class
     sample_weight : array-like of shape = [n_samples] or None, optional (default=None)
-            Weights of training data.
+        Weights of training data.
+    group : array-like
+        Group/query data, used for ranking task. sum(group) = n_samples.
```

Comment:
Can you expand this a bit more? I know that the first time I did learning-to-rank tasks on LightGBM, I was really confused about how to set up `group`.

```suggestion
        Group/query data, only used for ranking task. sum(group) = n_samples. For example,
            if you have a 100-record dataset with `group = [10, 20, 40, 10, 10]`, that means that you have
            5 groups, where the first 10 records are in the first group, records 11-30 are the second group, etc.
```

Reply:
Oh yeah it's definitely not particularly well-documented. This docstring was copied from the docs for LGBMRanker's `group` argument.

`group` is not a reference to a column index or anything, but a vector of group sizes. Your explanation in the suggested changes are spot-on.

At the end of the day, LightGBM has an internal check that looks something like `assert X.shape[0] == np.sum(group)`. Just giving a heads up that the DaskLGBMRanker's docs would now be different from LGBMRanker's docs.

Reply:
> Just giving a heads up that the DaskLGBMRanker's docs would now be different from LGBMRanker's docs.

Ah, thanks for that! I think I'll go make a separate PR for the non-Dask places this is documented.

Reply:
I added https://github.com/microsoft/LightGBM/pull/3772 to improve the docs on `group` in the Python and R packages.

Reply:
Slick! Addressed in d6e5209